### PR TITLE
Label/Priority and Filters

### PR DESCRIPTION
**Pull request by Megha-Joshi**
Features added in this PR :
- Labels added
- Priority added
- Filters added (according to label and priority)

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -8,6 +8,7 @@ import { Signup } from "./note-components/Authentication/signup";
 import { Route, Routes} from "react-router-dom";
 import { Trash } from "./note-components/Trash/trash";
 import { Archive } from "./note-components/Archive/archive";
+import { Label } from "./note-components/Label/label";
```
Comment (dev-prabhat) [inline src/App.js:11]: you can import all these components into index.js file and export it from there

File: src/index.js [hunk 2]
```diff
@@ -6,6 +6,7 @@ import { makeServer } from "./server";
 import { NoteProvider } from "./Context/note-context";
 import { AuthProvider } from "./Context/auth-context";
 import { BrowserRouter} from "react-router-dom";
+import { FilterProvider } from "./Context/filter-context";
```
Comment (dev-prabhat) [inline src/index.js:9]: import into the index.js file in the same directory and export it

**Review by dev-prabhat — APPROVED** [review]
Code is fine to me, given some suggestion 
You can merge this branch
Also if there is Preview link that would be nice , we can see the UI part also

File: src/Context/filter-context.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
+import { createContext, useReducer, useContext } from "react";
+import { useNote } from "./note-context";
+
+const FilterContext = createContext();
+
+
+const FilterProvider = ({children}) => {
+    const { noteState } = useNote();
+    const { note} = noteState;
+    console.log(note);
```
Comment (sanjitsarkar) [inline src/Context/filter-context.js:10]: remove console.log

File: src/note-components/Homepage/homepage.jsx [hunk 4]
```diff
@@ -5,23 +5,24 @@ import { Sidebar } from "../Sidebar/sidebar";
...
+const {note, priority} = noteState;
 const {authState} = useAuth();
 const { token } = authState;
-const [notes, setNote] = useState({ title: "", mainContent: "" });
 const navigate = useNavigate();
+const { filterState, finalFilter} = useFilter();
+
 
 const createNoteFunction = async () => {
 if (token) {
```
Comment (sanjitsarkar) [inline src/note-components/Homepage/homepage.jsx:22]: wrap all the routes that needs auth inside the protected/private route so you don't have to check for token every time you call some method need auth like this 
   ```
<Route element={<PrivateRoute />}>
          <Route path="/" element={<HomePage />} />
          <Route path="/bookmarks" element={<BookmarkedPage />} />
          <Route path="/explore" element={<ExplorePage />} />
          <Route path="/profile/:id" element={<ProfilePage />} />
          <Route path="/messages/" element={<MessagePage />} />
        </Route>
```

**Review by sanjitsarkar — APPROVED** [review]

File: src/note-components/Label/label.css [hunk 5]
```diff
@@ -0,0 +1,59 @@
+.chips-container {
+    height: 13.75rem;
+    border: 1px solid rgba(0, 0, 0, .2);
```
Comment (sumitk-singh) [inline src/note-components/Label/label.css:3]: use color variable here

File: src/note-components/Label/label.css [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    color: var(--font-color);
+    margin: 1.25rem;
+}
+
+
+.container-one {
+    display: flex;
+    background-color: #0000000a;
```
Comment (sumitk-singh) [inline src/note-components/Label/label.css:11]: better to use variable

File: src/note-components/Label/label.css [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+    background-color: var(--box-color);
+    border-radius: 10px;
+    margin-left: 10px;
+    padding: 5px 8px 5px 8px;
+    color: var(--font-color);
+    font-weight: 500;
+}
+
+.chips-btn:hover {
+    background-color: lightgrey;
```
Comment (sumitk-singh) [inline src/note-components/Label/label.css:48]: variable should be use

File: src/note-components/Label/label.jsx [hunk 8]
```diff
@@ -0,0 +1,52 @@
+import { Navbar } from "../Navbar/navbar";
+import { Sidebar } from "../Sidebar/sidebar";
```
Comment (sumitk-singh) [inline src/note-components/Label/label.jsx:2]: Better to use use index.js file in the same directory to import export these components

File: src/App.js [hunk 9]
```diff
@@ -19,6 +20,7 @@ function App() {
         <Route path="/signup" element={<Signup />} />
         <Route path="/trash" element={<Trash />} />
```
Comment (sumitk-singh) [inline src/App.js:21]: you may use the `Route` as per the private and public access.

**Review by sumitk-singh — APPROVED** [review]
Added some comment,
Although, rest of the code looks good to me. 
You may merge it after the little changes.

File: src/App.js (hunk 1 above)
Comment (Megha-Joshi) [reply src/App.js:11]: Okay will do it.

File: src/Context/filter-context.js (hunk 3 above)
Comment (Megha-Joshi) [reply src/Context/filter-context.js:10]: Okay

File: src/note-components/Homepage/homepage.jsx (hunk 4 above)
Comment (Megha-Joshi) [reply src/note-components/Homepage/homepage.jsx:22]: Okay! Thanks for explaining.

File: src/note-components/Label/label.css (hunk 5 above)
Comment (Megha-Joshi) [reply src/note-components/Label/label.css:3]: This color is getting used only 1 time. That's why I'm not using variable for it.

File: src/note-components/Label/label.jsx (hunk 8 above)
Comment (Megha-Joshi) [reply src/note-components/Label/label.jsx:2]: Okay

File: src/App.js (hunk 9 above)
Comment (Megha-Joshi) [reply src/App.js:21]: Okay will do it.

**Pull request merged**